week2/homework/src/index.js

```diff
@@ -1,3 +1,126 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const chalk = require('chalk');
+const yargs = require('yargs');
+const rl = require('readline').createInterface(process.stdin, process.stdout);
+const listToDos = require('./listToDos');
+const addToDo = require('./addToDo');
+const removeToDo = require('./removeToDo');
+const completeToDo = require('./completeToDo');
+const resetToDos = require('./resetToDos');
+const readToDo = require('./readToDo');
+const updateToDo = require('./updateToDo');
+const exportToDos = require('./exportToDos');
+const searchToDos = require('./searchToDos');
+
+yargs
+  .command({
+    command: 'list',
+    describe: 'List to-dos',
+    handler: function() {
+      listToDos();
+      rl.close();
+    },
+  })
+  .command({
+    command: 'add',
+    describe: 'Add a new to-do',
+    builder: {
+      complete: {
+        describe: 'To-do complete',
+        type: 'string',
+        default: '➖',
+      },
+    },
+    handler: function(argv) {
+      rl.question(chalk.yellow(`Please, enter a new to-do:\n`), answer => {
+        if (answer.length >= 5) {
+          addToDo(answer, argv.complete);
+          rl.close();
+        } else {
+          console.log(chalk.red('Please type at least 5 letters'));
+          this.handler(argv);
+        }
+      });
+    },
+  })
+  .command({
+    command: 'remove',
+    describe: 'Remove a to-do',
+    handler: function() {
+      rl.question(chalk.yellow(`Please, enter to-do order to remove:  `), answer => {
+        removeToDo(parseInt(answer));
+        rl.close();
+      });
+    },
+  })
+  .command({
+    command: 'complete',
+    describe: 'Complete a to-do',
+    handler: function() {
+      rl.question(chalk.yellow(`Please, enter to-do order to complete it:  `), answer => {
+        completeToDo(parseInt(answer));
+        rl.close();
+      });
+    },
+  })
+  .command({
+    command: 'reset',
+    describe: 'Remove all to-dos',
+    handler: function() {
```

Comment:
for the handlers too, it's one of the things we should be doing all the time, sometimes making it easier to debug and cleaner for the eye to read, we should move all these handlers to a isolated file that will take care of all question and lins we need to deal with.

then exporting them will give us the space here to see what kinf of commands we have with out running the help command from the command line, think about this: what if the `help` command is broken for some reason how can i know how your program works if I'm not a javascript person?

Reply:
I move it to separate file. Besides I add some methods to yargs, whether or not if the user typed anyother commands which are not included, it says to the user, that here is the help and usage of this app... :)